crates/hyperdrive-math/src/short/open.rs

```diff
@@ -85,75 +88,77 @@ impl State {
+
+            // Alice initializes the pool.
+            alice.initialize(fixed_rate, contribution, None).await?;
+
+            // Attempt to predict the spot price after opening a short.
+            let short_amount =
+                rng.gen_range(fixed!(0.01e18)..=bob.calculate_max_short(None).await?);
+            let current_state = bob.get_state().await?;
+            let expected_spot_price =
+                current_state.calculate_spot_price_after_short(short_amount, None);
+
+            // Open the short.
+            bob.open_short(short_amount, None, None).await?;
+
+            // Verify that the predicted spot price is equal to the ending spot
+            // price. These won't be exactly equal because the vault share price
+            // increases between the prediction and opening the short.
+            let actual_spot_price = bob.get_state().await?.calculate_spot_price();
+            let delta = if actual_spot_price > expected_spot_price {
+                actual_spot_price - expected_spot_price
+            } else {
+                expected_spot_price - actual_spot_price
             };
-            let max_iterations = 7;
-            let actual = panic::catch_unwind(|| {
-                state.calculate_max_short(
-                    U256::MAX,
-                    fixed!(0),
-                    checkpoint_exposure,
-                    None,
-                    Some(max_iterations),
-                )
-            });
-            match chain
-                .mock_hyperdrive_math()
-                .calculate_max_short(
-                    MaxTradeParams {
-                        share_reserves: state.info.share_reserves,
-                        bond_reserves: state.info.bond_reserves,
-                        longs_outstanding: state.info.longs_outstanding,
-                        long_exposure: state.info.long_exposure,
-                        share_adjustment: state.info.share_adjustment,
-                        time_stretch: state.config.time_stretch,
-                        vault_share_price: state.info.vault_share_price,
-                        initial_vault_share_price: state.config.initial_vault_share_price,
-                        minimum_share_reserves: state.config.minimum_share_reserves,
-                        curve_fee: state.config.fees.curve,
-                        flat_fee: state.config.fees.flat,
-                        governance_lp_fee: state.config.fees.governance_lp,
-                    },
-                    checkpoint_exposure,
-                    max_iterations.into(),
-                )
-                .call()
-                .await
-            {
-                Ok(expected) => {
-                    assert_eq!(actual.unwrap(), FixedPoint::from(expected));
-                }
-                Err(_) => assert!(actual.is_err()),
-            }
+            // TODO: Why can't this pass with a tolerance of 1e9?
+            let tolerance = fixed!(1e10);
```

Comment:
The spot price after long test uses `1e9`, but this one can't pass with the same tolerance... 🤔

Reply:
Maybe the source of the increase in error comes from the variable yield from the short?